packages/devtools_app/lib/src/profiler/cpu_profile_model.dart

```diff
@@ -16,82 +16,166 @@ import 'cpu_profile_transformer.dart';
 /// Data model for DevTools CPU profile.
 class CpuProfileData {
   CpuProfileData._({
-    @required this.stackFramesJson,
-    @required this.stackTraceEvents,
+    @required this.stackFrames,
+    @required this.cpuSamples,
     @required this.profileMetaData,
   }) {
     _cpuProfileRoot = CpuStackFrame(
-      id: 'cpuProfile',
-      name: 'all',
+      id: rootId,
+      name: rootName,
+      verboseName: rootName,
       category: 'Dart',
       url: '',
       profileMetaData: profileMetaData,
+      parentId: null,
     );
   }
 
   factory CpuProfileData.parse(Map<String, dynamic> json) {
+    final profileMetaData = CpuProfileMetaData(
+      sampleCount: json[sampleCountKey] ?? 0,
+      samplePeriod: json[samplePeriodKey],
+      stackDepth: json[stackDepthKey],
+      time: (json[timeOriginKey] != null && json[timeExtentKey] != null)
+          ? (TimeRange()
+            ..start = Duration(microseconds: json[timeOriginKey])
+            ..end = Duration(
+                microseconds: json[timeOriginKey] + json[timeExtentKey]))
+          : null,
+    );
+
+    // Initialize all stack frames.
+    final stackFrames = <String, CpuStackFrame>{};
+    final stackFramesJson =
+        jsonDecode(jsonEncode(json[stackFramesKey] ?? <String, dynamic>{}));
+    for (final MapEntry<String, dynamic> entry in stackFramesJson.entries) {
+      final stackFrameJson = entry.value;
+      final stackFrame = CpuStackFrame(
+        id: entry.key,
+        name: getSimpleStackFrameName(stackFrameJson[nameKey]),
+        verboseName: stackFrameJson[nameKey],
+        category: stackFrameJson[categoryKey],
+        // If the user is on a version of Flutter where resolvedUrl is not
+        // included in the response, this will be null. If the frame is a native
+        // frame, the this will be the empty string.
+        url: stackFrameJson[resolvedUrlKey] ?? '',
+        parentId: stackFrameJson[parentIdKey],
+        profileMetaData: profileMetaData,
+      );
+      stackFrames[stackFrame.id] = stackFrame;
+    }
+
+    // Initialize all CPU samples.
+    final stackTraceEvents =
+        (json[traceEventsKey] ?? []).cast<Map<String, dynamic>>();
+    final samples = stackTraceEvents
+        .map((trace) => CpuSample.parse(trace))
+        .toList()
+        .cast<CpuSample>();
```

Comment:
nit: why is this cast() required?

Reply:
without the cast, we get the error `type 'List<dynamic>' is not a subtype of type 'List<CpuSample>'`